### PR TITLE
Migrate delete to use go-sdk for library resource

### PR DESCRIPTION
**Pull request by edwardfeng-db**
## Changes

- Add `clusters_api_sdk.go` to implement same functionalities as in `clusters_api.go` with the go-sdk
- Migrate library delete to use the go sdk implementation

## Tests

- [ ] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Comment by edwardfeng-db** [comment]
@alexott Sorry for the confusion, I closed that one

File: clusters/clusters_api_sdk_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/terraform-provider-databricks/qa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
```
Comment (edwardfeng-db) [inline clusters/clusters_api_sdk_test.go:12]: These tests are the same as the ones in `clusters_api_test.go`, but with the `go-sdk`

File: clusters/clusters_api_sdk.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+// Start a terminated Spark cluster given its ID and wait till it's running
+func StartCluster(ctx context.Context, c *common.DatabricksClient, clusterID string) error {
+	_, err := StartClusterAndGetInfo(ctx, c, clusterID)
+	return err
+}
+
```
[the remark this answers is not in the document]
Comment (mgyucht) [reply clusters/clusters_api_sdk.go:17]: Is there really a need for both functions? Let's just have one; the caller can ignore the first returned value if it isn't relevant.

File: clusters/clusters_api_sdk.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// StartAndGetInfo starts cluster and returns info
+func StartClusterAndGetInfo(ctx context.Context, c *common.DatabricksClient, clusterID string) (*compute.ClusterDetails, error) {
```
Comment (mgyucht) [inline clusters/clusters_api_sdk.go:19]: Can we use WorkspaceClient here directly rather than `common.DatabricksClient`?

**Review by mgyucht — APPROVED** [review]
Couple small nits, otherwise LGTM!

File: clusters/clusters_api_sdk.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (edwardfeng-db) [reply clusters/clusters_api_sdk.go:17]: Yeah sounds good, we can start with just one. I created the wrapper to keep consistency with the http client code path but I guess it's not necessary for now

File: clusters/clusters_api_sdk.go (hunk 3 above)
Comment (edwardfeng-db) [reply clusters/clusters_api_sdk.go]: Updated

**Pull request merged**